### ISSUE TITLE
Use an alternative obscure url to circumvent common web censoring.

### ISSUE THREAD

**Issue by rradmin** [fetlife/android#160]
When my phone connects to the company wireless network the app doesn't work, because it's blocked on the firewall. I guess this issue comes up a few times when using public networks.

Since the url it connects to is trivial and used in the background, maybe you can code in some alternate urls for extra privacy. 

For example registering and  associating "vanillachat.io" to 151.101.64.64 and use that to connect.

**Comment by FernandoMiguel** [comment]
until adult filters are updated to include that one.
it's really not easy to work around that

best option, use a VPN.

**Comment by rradmin** [comment]
Most adult filters have manual verification, if a url is up for review people _visit_ it to take a look and classify it. You could make a white page with "coming soon" on that address. 

The thing I'm a bit concerned about: we need to protect the user. 

For the user accessing FL is an active act. Opening the site, or opening the app. Most users know when they want to connect because they know which networks are safe. 

 If you install the app, it syncs continuously, so that means if you use your company wireless your traffic can get flagged. Now, at this moment the only thing that give you away is the dns lookup. 

The ip the https connects to is the one from fastly. This cdn is used by many people so doesn't raise much flags. A separate ip en inconspicuous domain name should make it harder to track and to provide a higher privacy for users. 

At least you need to educate the user that installing the app equals to refreshing the FL site every minute. No point making a "vanilla" version of the app to give people a sense of privacy, while you're broadcasting their behavior over the web.

**Comment by kistitan** [comment]
@rradmin 
I think you have some good points, that we should think about, however this issue has a bit broader scope than Android development. 

@reneklacan @johnbaku 
As a summary as far as I got your suggestion is to get a different vanilla domain aside fetlife.com without any really html content and lead our Mobile API trough that domain.

I think it sound reasonable as it is valid argument not to have fetlife.com visible in the user traffic history that is most probably visible for the system administrators (sometimes even other employees) behind a company proxy.

On the other hand we should be careful as (based on my opinion at least) FetLife should not be able to be accused that we would like to avoid adult filters as sometimes they indeed do important jobs. 

Of course I still think it is a valid requirement we just need to discus it taking several point of view points into account. 


Side comment about the Vanilla version: We indeed should be careful not giving false privacy feelings to the users, but the main concerns the user have who requested the vanilla version was that they occasionally share their devices with other persons and they do not want them to accidentally or intentionally see what app they use.
Current vanilla version give help for these users at least even if it does not give protection for the scenario you described above.


As it is privacy related question I give medium priority to this issue to make sure we discuss it sooner than later.


Thanks a lot for your feedback and support.

**Comment by johnbaku** [comment]
@kistitan and @reneklacan I will leave this to you to decide with @rakvat. Just tell me what you guys decide.

**Comment by reneklacan** [comment]
@johnbaku 

I already talked about this with @kistitan in private and I think that alternative vanilla domain is just temporary solution and it's just matter of time when it will be discovered and maybe blocked even though it will serve just API. 

I would rather prefer to see support for configuring any proxy server within the app. It would require more effort from member's side to find some proxy provider or service but I think it's more scalable as we don't have to maintain list of vanilla domains.

**Comment by malachid** [comment]
Maybe you could create an IFTTT recipe to disable sync'ing while at work; thus still allowing it at home, munches, etc.

**Comment by GojuSuzi** [comment]
To be honest, I'd come down firmly against a dummy site, because aside from the constant need to update and avoid being caught in filters, it runs the risk in some countries of causing the site as a whole to get flagged as dodgy. For example, with UK's threatened age verification crackdown, any indication of even considering circumventing standard procedures could cause the site to be blocked via all means for the whole country. 

Better options would be along the lines of better proxy support, or even a toggle option to block background traffic either entirely or on a specific list of networks (maybe on Vanilla only?) so that you can prevent it leaving footprints when you connect to networks where you're monitored and only work if you actively open the app and/or when connected on mobile data/WiFi where no one will be snooping.

**Comment by restrainedimages** [comment]
I'm totally against trying to avoid filters as I think that is unethical. The WIFI provider has effectively made it clear they don't want their network used to access adult content. Going round that would be a consent violation and against everything we stand for.

In the short term I think you should...
1) Ensure the app doesn't communicate if you aren't logged in. This gives users one way to prevent the app from appearing in logs on monitored WIFI connections.
2) Add a warning message on the login page that explains the situation / risk / work-around.

In the long term I think the option to disable connectivity when connected to certain WIFI networks would be an excellent solution. But once above is done that could be a lower priority.

**Comment by kistitan** [comment]
I like these suggestions.
I will make sure #1 will be added soon.
#2 is also important we should add it to our terms and conditions/general app description screen, that is not ready yet.

#3 is also nice suggestion we should add it to some advance settings.

**Comment by DemiMarie** [comment]
For users who have Orbot (Tor for Android) installed, would it be possible to cooperate with Orbot to route all app traffic through Tor?  That really would solve the problem, since Tor is untraceable.

**Comment by malachid** [comment]
From the Guardian Project:

PRIVACY FOR APPS: Any installed app can use Tor if it has a proxy feature,
using the settings found here: https://goo.gl/2OA1y
<http://www.google.com/url?q=https://goo.gl/2OA1y&sa=D&usg=AFQjCNFsNchgD26l7wuGhljFAZ_bG9ZpDQ>

**Comment by Mikaela** [comment]
The link above goes to https://guardianproject.info/apps/orbot/ .

Are there plans to bring proxy settings or Orbot support to the app?

If Orbot becomes supported, would it also be possible for you to run a hidden service to remove the need of trusting the Tor exit nodes?
